runtime/runtime/src/lib.rs

```diff
@@ -1645,14 +1653,19 @@ impl Runtime {
         validator_proposals: &mut Vec<ValidatorStake>,
     ) -> Result<(), RuntimeError> {
         let local_processing_start = std::time::Instant::now();
-        let local_receipt_count = processing_state.local_receipts.len();
+        let local_receipts = std::mem::take(&mut processing_state.local_receipts);
+        let local_receipt_count = local_receipts.len();
         if let Some(prefetcher) = &mut processing_state.prefetcher {
             // Prefetcher is allowed to fail
-            let (front, back) = processing_state.local_receipts.as_slices();
+            let (front, back) = local_receipts.as_slices();
             _ = prefetcher.prefetch_receipts_data(front);
             _ = prefetcher.prefetch_receipts_data(back);
         }
-        while let Some(receipt) = processing_state.next_local_receipt() {
+
+        let mut local_receipt_iter = local_receipts.iter();
+        let mut prep_lookahead_iter = local_receipts.iter();
+        let _ = prep_lookahead_iter.next(); // stagger preparation to not prepare the first receipt.
```

Comment:
what if the first receipt contains deploy contract action, should we block the account_id for further pipelining in that case?

Reply:
Good eye! I think I addressed this by not trying to be as smart. I should add a regression test for this too (please do not resolve this thread quite yet.)

@Ekleog-NEAR I also made changes which I think will make it clearer when and how many receipts are submitted in parallel, so the currently last commit should resolve your concerns with that too?

Reply:
The changes LGTM :)